src/coreclr/scripts/superpmi.py

```diff
@@ -659,11 +659,18 @@ def __init__(self, coreclr_args):
             # Therefore, we produce a copy of the JIT binary for SuperPMI to use. 
             jit_name_ext = os.path.splitext(jit_name)[1]
             jit_name_without_ext = os.path.splitext(jit_name)[0]
-            self.superpmi_jit_path = os.path.join(coreclr_args.core_root, jit_name_without_ext + "_superpmi." + jit_name_ext)
-            shutil.copyfile(self.jit_path, self.superpmi_jit_path)
+            self.superpmi_jit_path_temp_dir = tempfile.TemporaryDirectory(ignore_cleanup_errors=True)
```

Comment:
There is already a temporary directory created inside `collect` so I think you should move this logic to create the copy into `__collect_mc_files__` and use that temporary directory. It will also allow you to get rid of the cleanup of the copied JIT.

Reply:
That seems to be better. done